securedrop_client/gui/conversation/export/device.py

```diff
@@ -129,13 +115,74 @@ def print_file(self, file_uuid: str) -> None:
         if not self._controller.downloaded_file_exists(file):
             return
 
-        if not self._controller.qubes:
-            return
-
         self.print_requested.emit([file_location])
 
     def _move_export_service_to_thread(self, thread: QThread) -> None:
         self._export_service_thread = thread
 
         self._export_service.moveToThread(self._export_service_thread)
         self._export_service_thread.start()
+
+
+class StubDevice(QObject):
+    """Abstracts an export service for use in GUI components.
+
+    This class allows to emulate exports without side effects beyond the GUI.
+    """
+
+    export_preflight_check_succeeded = pyqtSignal()
+    export_succeeded = pyqtSignal()
+
+    print_preflight_check_succeeded = pyqtSignal()
+
+    def __init__(
+        self, controller: Controller, export_service_thread: Optional[QThread] = None
+    ) -> None:
+        super().__init__()
+
+        self._controller = controller
+
+    def run_printer_preflight_checks(self) -> None:
+        """
+        Run preflight checks to make sure the Export VM is configured correctly.
+        """
+        logger.info("Running printer preflight check")
+
+        self.print_preflight_check_succeeded.emit()
+
+    def run_export_preflight_checks(self) -> None:
+        """
+        Run preflight checks to make sure the Export VM is configured correctly.
+        """
+        logger.info("Running export preflight check")
+
+        self.export_preflight_check_succeeded.emit()
+
+    def export_file_to_usb_drive(self, file_uuid: str, passphrase: str) -> None:
+        """
+        Send the file specified by file_uuid to the Export VM with the user-provided passphrase for
+        unlocking the attached transfer device.  If the file is missing, update the db so that
+        is_downloaded is set to False.
+        """
+        file = self._controller.get_file(file_uuid)
+        file_location = file.location(self._controller.data_dir)
+        logger.info("Exporting file in: {}".format(os.path.dirname(file_location)))
+
+        if not self._controller.downloaded_file_exists(file):
+            return
+
+        self.export_succeeded.emit()
+
+    def print_file(self, file_uuid: str) -> None:
+        """
+        Send the file specified by file_uuid to the Export VM. If the file is missing, update the db
+        so that is_downloaded is set to False.
```

Comment:
These comments are not too accurate... but we're not at the point where we can say: "Prints some logs and does nothing else." because of the side effects of `controller.downloaded_file_exists`. So I'll keep this docstring for now, until we understand better how much of that we can remove.

Reply:
I removed them in #1541, and we can add them back when useful.